### ISSUE TITLE
no-ignored-initial-value reports false positives when using computed property names

### ISSUE THREAD

**Issue by karol-majewski** [SonarSource/SonarTS#682]
Hello and thank you for your work.

**I want to report a bug.**

**SonarTS version:** 1.7.0<br>
**Node.js version:** 10.7.0<br>
**TypeScript version:** 2.9.2<br>
**TSLint version:** . 5.11.0<br>

**Rule key:** `no-ignored-initial-value  `

**Reproducer** 
```ts
type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;

export const omit = <T, K extends keyof T>(source: T, property: K): Omit<T, K> => {
  const { [property]: _, ...rest } = source as any;

  return rest as Omit<T, K>;
};
```

**Logs**
```
Introduce a new variable or use its initial value before reassigning "property".
```

**Expected behavior**

No error. `property` is used to compute a property which is going to be picked from `source`.

**Comment by vilchik-elena** [comment]
Issue should be fixed in symbol table. The problem also affects dead store rule:
```ts
function foo(obj: any) {
  let prop = "prop";
  const { [prop]: _, ...rest } = obj;
  return rest;
}
```
